daemon/cmd/fqdn_test.go

```diff
@@ -121,7 +128,20 @@ func (ds *DaemonFQDNSuite) SetUpTest(c *C) {
 	})
 	d.endpointManager = WithCustomEndpointManager(&dummyEpSyncher{})
 	d.policy.GetSelectorCache().SetLocalIdentityNotifier(d.dnsNameManager)
+	d.ipcache = ipcache.NewIPCache(&ipcache.Configuration{
```

Comment:
It's a good idea, but it will require refactoring the `Daemon` struct to use an interface representing the ipcache, because currently it uses a direct pointer to the real ipcache type.

Can do it as a follow up.

Reply:
https://github.com/cilium/cilium/issues/22524